aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncMocking.java

```diff
@@ -248,5 +249,25 @@ public SyncConfigurator mockSuccessResponses() {
             );
             return SyncConfigurator.this;
         }
+
+        /**
+         * Triggers an exception when invoking the sync method.
+         * @param <T> Type of models for which a response is mocked
+         * @return The same Configurator instance, to enable chaining of calls
+         */
+        public <T extends Model> SyncConfigurator mockFailure() {
+            doAnswer(invocation -> {
+                final int errorConsumerPosition = 3;
+                final Consumer<DataStoreException> consumer = invocation.getArgument(errorConsumerPosition);
+                consumer.accept(new DataStoreException("Something timed out during sync.", "Nothing to do."));
+                return new NoOpCancelable();
+            }).when(appSync).sync(
+                any(), // Item class to sync
+                any(), // last sync time
+                any(), // Consumer<Iterable<ModelWithMetadata<T>>>
```

Comment:
Do you have `anyConsumer()` available here? I forget where that utility lives.

Reply:
It's a private static function buried in [SubscriptionProcessorTest](https://github.com/aws-amplify/amplify-android/blob/6a4848025c79e928a9ed3d69267a924628241188/aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessorTest.java#L231) It probably should be moved elsewhere.